[SDCT-837] Document new Test Impact Analysis limitations (#38631)
DataDog/documentation: content/en/tests/test_impact_analysis/troubleshooting/_index.md

### Fewer tests skipped than expected

> There are edge-cases where Test Impact Analysis will skip less tests than expected.

Reviewer: Three issues in one sentence:

- `edge-cases` -> `edge cases` (not hyphenated as a noun)
- `will skip` -> present tense; Vale flags `will` as a temporal word (`Datadog.tense`)
- `less tests` -> `fewer tests` ("tests" is a count noun)

```suggestion
There are edge cases where Test Impact Analysis skips fewer tests than expected.
```

> - A very large git history may result in slightly fewer tests being skipped, because Datadog analyzes up to the first 100 recent commits.

Reviewer: "the first 100 recent commits" is self-contradictory — "first" and "recent" pull in opposite directions. The main page has it right ("the 100 most recent commits"). Also `git` -> `Git` per the word list (capitalized except in commands).

```suggestion
- A very large Git history may result in slightly fewer tests being skipped, because Datadog analyzes up to the 100 most recent commits.
```

> - A commit that changed more than 5,000 files is not analyzed for impact, so it is expected the coverage information of that commit to be ignored.

Reviewer: "it is expected the coverage information ... to be ignored" isn't a valid construction, and the hedge adds nothing — the information *is* ignored. The tense also doesn't match the sibling bullets ("changed" here vs. present tense elsewhere).

```suggestion
- A commit that changes more than 5,000 files is not analyzed for impact, so the coverage information from that commit is ignored.
```

- A test or suite that covers more than 16,000 files is never skipped.

> - A very small fraction of tests (well under one percent) may run even when they could have been skipped.

Reviewer: Percentages take numerals and the `%` symbol rather than spelled-out words.

```suggestion
- A very small fraction of tests (well under 1%) may run even when they could have been skipped.
```

export DD_TESTOPTIMIZATION_TIA_TEST_SKIPPING_MODE=suite

> {{< /code-block >}}

Author: remove this text, it is no longer true

> Other situations can also lead to fewer skipped tests. In all of them the extra tests run safely: Test Impact Analysis never skips a test that should have run.

Author: ```suggestion
There are edge-cases were Test Impact Analysis will skip less tests than expected.
```

> - A very large changeset may result in slightly fewer tests being skipped, because Datadog analyzes up to the 100 most recent commits.

Author: ```suggestion
- A very large git history may result in slightly fewer tests being skipped, because Datadog analyzes up to the first 100 recent commits.
```

> - A single commit that changes more than 5,000 files is not analyzed for impact, so its tests run rather than being skipped.

Author: ```suggestion
- A commit that changed more than 5,000 files is not analyzed for impact, so it is expected the coverage information of that commit to be ignored.
```

> - A test or suite that covers more than 16,000 files is not skipped.

Author: ```suggestion
- A test or suite that covers more than 16,000 files is never skipped.
```

- A very small fraction of tests (well under one percent) may run even when they could have been skipped.

For more detail on these behaviors, see [Limitations and expected behavior][7].

### Code coverage is incomplete or missing

> Per-test (line-level) code coverage is collected for Java, JavaScript, .NET, and Go. It is not collected for Ruby, Python, or Swift. For very large test suites, per-test coverage may occasionally be unavailable even in supported languages, and affected tests are marked with a missing-coverage indicator. Suite-level coverage remains reliable and is what Test Impact Analysis uses to decide which tests to skip.

Author: This is all wrong. First of all, this should be inside the "Code Coverage Backfilling" section. Second, the feature that may be degraded is the code coverage backfill (i.e.: backfill the test coverage of tests skipped by TIA in the code coverage report of the run)
